### PR TITLE
1817/1867/18CO: Make common the tile lay restriction that

### PR DESCRIPTION
**Pull request by jenf**
you cannot upgrade the same hex.

Also make the UI say if you can upgrade/lay at that point

### CODE REVIEW

File: lib/engine/step/tracker.rb [hunk 1]
```diff
@@ -26,6 +27,7 @@ def get_tile_lay(entity)
         action[:lay] = !@upgraded if action[:lay] == :not_if_upgraded
         action[:upgrade] = !@upgraded if action[:upgrade] == :not_if_upgraded
         action[:cost] = action[:cost] || 0
+        action[:upgrade_same_hex] = true if action[:upgrade_same_hex].nil?
```
Comment (tobymao) [inline lib/engine/step/tracker.rb:30]: unless action[:upgrade}

**Review by tobymao — APPROVED** [review]

**Comment by ryandriskel** [comment]
Nice find Jen!

**Comment by jenf** [comment]
Don't merge this checking an issue

File: lib/engine/step/tracker.rb (hunk 1 above)
Comment (jenf) [reply lib/engine/step/tracker.rb]: that doesn't work since action[:upgrade_same_hex] unless i flip the tense to cannot_reuse_same_hex

Comment (tobymao) [reply lib/engine/step/tracker.rb]: ok no problem this is fine then

**Review by tobymao — APPROVED** [review]

File: lib/engine/step/tracker.rb (hunk 1 above)
Comment (jenf) [reply lib/engine/step/tracker.rb]: I'd already done it, which would you prefer?

**Comment by tobymao** [comment]
don't care

**Pull request merged**